lib/packages/shared-types/opensearch/main/transforms/legacy-transforms/initial-waiver.ts

```diff
@@ -0,0 +1,18 @@
+import { events } from "shared-types";
+import { baseTransform } from "./common-legacy-transform";
+
+export const transform = () => {
+  return events["legacy-event"].legacyEventSchema.transform((data) => {
+    // Use the shared transformation logic
+    const baseResult = baseTransform(data);
```

Comment:
Should you be able to move `baseTransform`'s logic into a `.transform`?

```ts
export const legacyEventSchema = legacySharedSchema
                                  .merge(...)
                                  .transform(/* baseTransform logic goes here */)
```

Reply:
I dont have any objection against doing this but what is the benefit of having a merged transform vs a shared method in each transform?

Reply:
You've repeated the logic in multiple files when it's possible to avoid that. So why not keep things DRY?

Reply:
Apologies but either way you are either calling an independent function or calling a merge function from the other transform. So DRY doesnt apply as both are equal. I will concede the built in merge(...).transform() is more concise and easier to digest so will implement this suggestion.

BTW, not trying to be contrary and I do appreciate the feedback! I just like to understand intent. :)